prover/src/lib.rs

```diff
@@ -152,6 +159,13 @@ pub trait Prover {
     /// proof size, and proof security level.
     fn options(&self) -> &ProofOptions;
 
+    /// TODO: docs
+    fn new_trace_lde<E: FieldElement<BaseField = Self::BaseField>>(
+        &self,
+        trace_info: &TraceInfo,
+        blowup: usize,
+    ) -> Self::TraceLde<E>;
```

Comment:
Looking at this now, I do wonder if we should instead replace the `TraceLde::set_main_trace()` with `TraceLde::new()` function which would look something like this:
```Rust
fn new(
    trace_info: &TraceInfo,
    main_trace: &ColMatrix<Self::BaseField>,
    domain: &StarkDomain<Self::BaseField>,
) -> Self
```
And then call this constructor instead of using `new_trace_lde()`.

This would get rid of all the annoying checks for whether the main trace has already been set. And if we wanted to pass in some custom info into the `new()` function, we could still do it via the `meta` property of `TraceInfo`.

If we do this, we would also need to add `get_main_trace_commitment()` method to `TraceLde` - but I think that shouldn't cause any issues.

Reply:
I've updated this, but I have a question - we now need to store the trace polynomials so that they can be retrieved during `get_main_trace_commitment`, and it gets cloned in that method. I wonder if it would be better to instead do something else, like returning the `TracePolyTable` from `TraceLde::new` as well

Reply:
Yes - I like this approach better. So, basically, `TraceLde::new()` would look like this, right?
```Rust
fn new(
    trace_info: &TraceInfo,
    main_trace: &ColMatrix<Self::BaseField>,
    domain: &StarkDomain<Self::BaseField>,
) -> (Self, TracePolyTable<Self::BaseField>)
```

I guess another approach is to move the management of `TracePolyTable` completely into `TraceLde` - but I think this is a bigger refactoring and if we do want to do it, we could do it in a different PR.

Reply:
> I guess another approach is to move the management of TracePolyTable completely into TraceLde - but I think this is a bigger refactoring and if we do want to do it, we could do it in a different PR.

I considered this, but it introduces other complications, since we would still have to do cloning to initialize the DEEP composition poly. I think the simpler approach (returning `TracePolyTable`) will be cleaner